apm-sniffer/apm-sdk-plugin/mongodb-2.x-plugin/src/main/java/org/skywalking/apm/plugin/mongodb/v2/MongoDBV2MethodInterceptor.java

```diff
@@ -0,0 +1,111 @@
+ * limitations under the License.
+ *
+ * Project repository: https://github.com/OpenSkywalking/skywalking
+ */
+
+package org.skywalking.apm.plugin.mongodb.v2;
+
+import com.mongodb.CommandResult;
+import com.mongodb.Mongo;
+import com.mongodb.ServerAddress;
+import com.mongodb.DB;
+import com.mongodb.WriteResult;
+import com.mongodb.AggregationOutput;
+import java.lang.reflect.Method;
+import java.util.List;
+import org.skywalking.apm.agent.core.context.ContextCarrier;
+import org.skywalking.apm.agent.core.context.ContextManager;
+import org.skywalking.apm.agent.core.context.tag.Tags;
+import org.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.skywalking.apm.network.trace.component.ComponentsDefine;
+
+/**
+ * {@link MongoDBV2MethodInterceptor} intercept method of {@link com.mongodb.DBCollection#find()}
+ * or{@link com.mongodb.DBCollectionImpl#insertImpl}.... record the mongoDB host, operation name ...
+ */
+
+public class MongoDBV2MethodInterceptor implements InstanceMethodsAroundInterceptor, InstanceConstructorInterceptor {
+
+    private static final String DB_TYPE = "MongoDB";
+
+    private static final String MONGO_DB_OP_PREFIX = "MongoDB/";
+
+    @Override public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+        Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+
+        String remotePeer = (String)objInst.getSkyWalkingDynamicField();
+        AbstractSpan span = ContextManager.createExitSpan(MONGO_DB_OP_PREFIX + method.getName(), new ContextCarrier(), remotePeer);
+        span.setComponent(ComponentsDefine.MONGODB);
+        Tags.DB_TYPE.set(span, DB_TYPE);
+        SpanLayer.asDB(span);
+
+    }
+
+    @Override public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+        Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        AbstractSpan activeSpan = ContextManager.activeSpan();
+        if (ret instanceof WriteResult) {
+            WriteResult wresult = (WriteResult)ret;
+            if (!wresult.getCachedLastError().ok()) {
+                Tags.STATUS_CODE.set(activeSpan, "failed");
+            }
+        } else if (ret instanceof CommandResult) {
+            CommandResult cresult = (CommandResult)ret;
+            if (!cresult.ok()) {
+                Tags.STATUS_CODE.set(activeSpan, "failed");
+            }
+        } else if (ret instanceof AggregationOutput) {
+            AggregationOutput aresult = (AggregationOutput)ret;

```

Comment:
The result value of `wresult.getCachedLastError()` and `aresult.getCommandResult()` is the `CommandResult ` class. I think the logic should be more simply